packages/vscode-dependencies-validation/src/extension.ts

```diff
@@ -41,11 +110,52 @@ async function displayProblematicDependencies(
 ): Promise<void> {
   const start = Date.now();
 
-  const problemDeps: NPMDependencyIssue[] = await findDependencyIssues(
+  const npmLsResult: NpmLsResult = await findDependencyIssues(
     packageJsonUri.fsPath
   );
 
   void window.showInformationMessage(
-    `found ${problemDeps.length} problems in ${Date.now() - start} milliseconds`
+    `found ${npmLsResult.problems?.length || 0} problems in ${
+      Date.now() - start
+    } milliseconds`
+  );
+}
+
+function subscribeToDocumentChanges(
```

Comment:
this is based on vscode extension sample?

Reply:
yes